src/WebJobs.Script.Scaling/ArgumentUtils.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.Azure.WebJobs.Script.Scaling
+{
+    [SuppressMessage("Microsoft.Naming", "CA1704:IdentifiersShouldBeSpelledCorrectly", Justification = "By design")]
+    public static class ArgumentUtils
+    {
+        public static void ValidateArgument(string name, string arg)
+        {
+            if (string.IsNullOrEmpty(arg))
+            {
+                throw new ArgumentNullException(name, "Parameter must not be null or empty.");
+            }
+        }
+
+        public static void ValidateArgument(string name, object arg)
+        {
+            if (arg == null)
+            {
+                throw new ArgumentNullException(name, "Parameter must not be null.");
+            }
+        }
+
+        public static void ValidateArgument(IWorkerInfo worker)
+        {
+            ValidateArgument("worker", (object)worker);
+            ValidateArgument("worker.SiteName", worker.SiteName);
+            ValidateArgument("worker.StampName", worker.StampName);
+            ValidateArgument("worker.WorkerName", worker.WorkerName);
+        }
+
+        public static void ValidateActivityId(string arg)
+        {
+            Guid guid;
+            if (!Guid.TryParse(arg, out guid))
+            {
+                throw new ArgumentException("Parameter 'activityId' must be a valid guid.");
```

Comment:
This message seems off, since it isn't clear what 'activityId' parameter it is referring to. Perhaps reword to "Invalid Activity Id" or something?